### PR TITLE
some fixes and stuff :)

### PR DESCRIPTION
**Pull request by slbsh**
- **fix: copy README from aur_release branch**
- **change: add yes/no as valid bool values (ssh config aah)**
- **change: better config file error fallback handling**
- **chore: make formatting a bit more consistent**

### CODE REVIEW

File: config.c [hunk 1]
```diff
@@ -35,16 +36,22 @@ write_default_config(FILE *f)
 }
 
 Config
-load_config(const char *fileName)
+load_config(const char *filename)
```
Comment (nick-bors) [inline config.c]: shouldn't this be file_name? or is filename one word idk

File: config.c [hunk 2]
```diff
@@ -35,16 +36,22 @@ write_default_config(FILE *f)
...
-        f = fopen(fileName, "wr");
+    // NOTE: reading a file can fail for many reasons. 
+    // we should not be trying to create a new one in most of em
+	 if (access(filename, F_OK | R_OK)) {
+        printf("Config file %s does not exist, creating default\n", filename);
+        FILE *f = fopen(filename, "wr");
         if (f == NULL)
-            die("Unable to open config file %s\n", fileName);
+            die("Unable to open config file %s\n", filename);
         write_default_config(f);
```
Comment (nick-bors) [inline config.c]: Maybe you should close between opening with write-read and opening with just read

File: config.c [hunk 3]
```diff
@@ -35,16 +36,22 @@ write_default_config(FILE *f)
...
+load_config(const char *filename)
 {
-    FILE *f = fopen(fileName, "r");
-    if (f == NULL) {
-        f = fopen(fileName, "wr");
+    // NOTE: reading a file can fail for many reasons. 
+    // we should not be trying to create a new one in most of em
+	 if (access(filename, F_OK | R_OK)) {
+        printf("Config file %s does not exist, creating default\n", filename);
+        FILE *f = fopen(filename, "wr");
```
Comment (nick-bors) [inline config.c]: if youre only writing a default config and re-opening it in readonly mode later anyways, why not just open as 'w'?

File: config.c [hunk 4]
```diff
@@ -60,36 +67,38 @@ load_config(const char *fileName)
...
                     break;
                 case 't':
                 case 'T':
+                case 'y':
+                case 'Y':
                 case '1':
                     conf.windowed = true;
                     break;
                 default:
                     break;
```
Comment (nick-bors) [inline config.c]: This logic is separated into parse_bool in master, so when i merge ill move it there 🤝🤝

File: main.c [hunk 5]
```diff
@@ -190,9 +191,9 @@ keypress(XEvent *e)
 void
 scroll_up(unsigned int delta, bool fl_enabled)
 {
-    if ((delta > 0) && fl_enabled) {
+    if (delta > 0 && fl_enabled)
         flashlight.delta_radius += 250.0f;
-    } else {
```
Comment (nick-bors) [inline main.c]: Id prefer for nested logic with an else clause to be wrapped in braces for easier readability.

File: config.c [hunk 6]
```diff
@@ -60,36 +67,38 @@ load_config(const char *fileName)
...
+            else if (!strcmp(arg, "drag_friction"))
                 conf.drag_friction = strtof(c, NULL);
-            } else if (!strcmp(arg, "scale_friction")) {
+            else if (!strcmp(arg, "scale_friction"))
                 conf.scale_friction = strtof(c, NULL);
-            } else if (!strcmp(arg, "key_move_speed")) {
+            else if (!strcmp(arg, "key_move_speed"))
                 conf.key_move_speed = strtof(c, NULL);
-            } else if (!strcmp(arg, "windowed")) {
+            else if (!strcmp(arg, "windowed"))
```
Comment (nick-bors) [inline config.c]: nice formatting change actually!

File: main.c [hunk 7]
```diff
@@ -201,9 +202,9 @@ scroll_up(unsigned int delta, bool fl_enabled)
 void
 scroll_down(unsigned int delta, bool fl_enabled)
 {
-    if ((delta > 0) && fl_enabled) {
+    if (delta > 0 && fl_enabled)
         flashlight.delta_radius -= 250.0f;
-    } else {
+    else {
```
Comment (nick-bors) [inline main.c]: Nested logic (see comment above)

File: main.c [hunk 8]
```diff
@@ -261,14 +262,13 @@ motion_notify(XEvent *e)
 }
 
 int
-main(int argc, char *argv[])
```
Comment (nick-bors) [inline main.c]: At some point ill add these back for maybe some cli options but for now removing them gets rid of warnings so this is good

File: navigation.c [hunk 9]
```diff
@@ -21,11 +21,10 @@ update_flashlight(Flashlight *fl, float dt)
     }
 
     /* Smoothly interpolate between on/off */
-    if (fl->is_enabled) {
+    if (fl->is_enabled)
         fl->shadow = MIN(fl->shadow + SHADOW_ACCEL * dt , MAX_SHADOW);
-    } else {
```
Comment (nick-bors) [inline navigation.c]: same thing as before

File: navigation.c [hunk 10]
```diff
@@ -67,7 +66,7 @@ update_camera(Camera *cam, Config *config, Mouse *mouse, Vec2f window_size)
         cam->delta_scale -= cam->delta_scale * cam->dt * config->scale_friction;
     }
 
-    if (!mouse->dragging && (LEN(cam->velocity) > VELOCITY_THRESHOLD)) {
+    if (!mouse->dragging && LEN(cam->velocity) > VELOCITY_THRESHOLD) {
```
Comment (nick-bors) [inline navigation.c]: Yeah I originally had these because my LEN macro was bad but I since fixed it and forgot to update

**Review by nick-bors — DISMISSED** [review]
Some changes as described but overall really good cleanup (especially with whitespaces and such). Thanks Anthony!

File: config.c (hunk 1 above)
Comment (slbsh) [reply config.c]: rtfm `man fopen`

File: config.c (hunk 2 above)
Comment (slbsh) [reply config.c]: yea good catch

File: config.c (hunk 4 above)
Comment (slbsh) [reply config.c]: 💀

**Pull request merged**